utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,597 @@
+        ml_model_status = ml_client.get_model_info(model_id)
+        assert ml_model_status.get("model_state") == "UNDEPLOYED"
+    except Exception as e:
+        assert False, f"Raised Exception in {model_format} model undeployment: {e}"
+
+    # 5.) Delete the model
+    try:
+        delete_model_obj = ml_client.delete_model(model_id)
+        assert delete_model_obj.get("result") == "deleted"
+    except Exception as e:
+        assert False, f"Raised Exception in deleting {model_format} model: {e}"
+
+    # 6.) Return embedding outputs for model verification
+    return embedding_data
+
+
+def verify_embedding_data(
+    original_embedding_data: List["DTypeLike"],
+    tracing_embedding_data: List["DTypeLike"],
+) -> bool:
+    """
+    Verify the embeddings generated by the traced model with those of original model
+
+    :param original_embedding_data: Embedding outputs of TEST_SENTENCES generated by the original model
+    :type original_embedding_data: List['DTypeLike']
+    :param tracing_embedding_data: Embedding outputs of TEST_SENTENCES generated by the traced model
+    :type tracing_embedding_data: List['DTypeLike']
+    :return: Whether the embeddings generated by the traced model match with those of original model
+    :rtype: bool
+    """
+    failed_cases = []
+    for i in range(len(TEST_SENTENCES)):
+        try:
+            np.testing.assert_allclose(
+                original_embedding_data[i],
+                tracing_embedding_data[i],
+                rtol=RTOL_TEST,
+                atol=ATOL_TEST,
+            )
+        except Exception as e:
+            failed_cases.append((TEST_SENTENCES[i], e))
+
+    if len(failed_cases):
+        print(
+            "\nOriginal embeddings DOES NOT matches the embeddings in the following case(s):"
+        )
+        for sentence, e in failed_cases:
+            print(sentence)
+            print(e)
+        return False
+    else:
+        return True
+
+
+def prepare_files_for_uploading(
+    model_id: str,
+    model_version: str,
+    model_format: str,
+    src_model_path: str,
+    src_model_config_path: str,
+) -> None:
+    """
+    Prepare files for uploading by storing them in UPLOAD_FOLDER_PATH
+
+    :param model_id: Model ID of the pretrained model
+    :type model_id: string
+    :param model_version: Version of the pretrained model for registration
+    :type model_version: string
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    :param src_model_path: Path to model files for uploading
+    :type src_model_path: string
+    :param src_model_config_path: Path to model config files for uploading
+    :type src_model_config_path: string
+    :return: Tuple of dst_model_path (path to model zip file) and dst_model_config_path
+    (path to model config json file) in the UPLOAD_FOLDER_PATH
+    :rtype: Tuple[str, str]
+    """
+    model_name = str(model_id.split("/")[-1])
```

Comment:
we can keep the 1st one also for the library name which can be used in `dst_model_filename` instead of putting string `sentence-transformers`

Reply:
Sure! That's a great idea.